tests/Integration/Behaviour/Features/Context/Domain/Product/ProductCombinationFeatureContext.php

```diff
@@ -0,0 +1,136 @@
+
+namespace Tests\Integration\Behaviour\Features\Context\Domain\Product;
+
+use Behat\Gherkin\Node\TableNode;
+use PHPUnit\Framework\Assert;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Command\GenerateProductCombinationsCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Query\GetProductCombinationsForEditing;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\QueryResult\CombinationForEditing;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\QueryResult\ProductCombinationsForEditing;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductException;
+use Tests\Integration\Behaviour\Features\Context\Util\PrimitiveUtils;
+
+class ProductCombinationFeatureContext extends AbstractProductFeatureContext
+{
+    /**
+     * @When I generate combinations for product :productReference using following attributes:
+     *
+     * @param string $productReference
+     * @param TableNode $table
+     */
+    public function generateCombinations(string $productReference, TableNode $table): void
+    {
+        $tableData = $table->getRowsHash();
+        $groupedAttributeIds = $this->parseGroupedAttributeIds($tableData);
+
+        try {
+            $this->getCommandBus()->handle(new GenerateProductCombinationsCommand(
+                $this->getSharedStorage()->get($productReference),
+                $groupedAttributeIds
+            ));
+        } catch (ProductException $e) {
+            $this->setLastException($e);
+        }
+    }
+
+    /**
+     * @Then product :productReference should have following combinations:
+     *
+     * @param string $productReference
+     * @param TableNode $table
+     */
+    public function assertProductCombinations(string $productReference, TableNode $table): void
+    {
+        $dataRows = $table->getColumnsHash();
+        $expectedCombinationsCount = count($dataRows);
+
+        /** @var ProductCombinationsForEditing $combinationsForEditing */
+        $combinationsForEditing = $this->getQueryBus()->handle(new GetProductCombinationsForEditing(
+            $this->getSharedStorage()->get($productReference),
+            $this->getDefaultLangId(),
+            $expectedCombinationsCount
+        ));
+
+        Assert::assertEquals(
+            count($combinationsForEditing->getCombinations()),
+            $expectedCombinationsCount,
+            'Unexpected combinations count'
+        );
+
+        /** @var CombinationForEditing $combinationForEditing */
+        foreach ($combinationsForEditing as $key => $combinationForEditing) {
+            Assert::assertEquals(
+                $dataRows[$key]['combination name'],
+                $combinationForEditing->getCombinationName(),
+                'Unexpected combination'
+            );
+        }
+    }
+
+    /**
+     * @param string $groupsAttributesReferencesPairs
+     *
+     * @return array
+     */
+    private function parseAttributes(string $groupsAttributesReferencesPairs): array
```

Comment:
Seems like this method is not used, was it designed to check the Combination attributes after generation?
Although this check might be more accurate, the name comparison seems fine for now You can delete this one if you don't use it

Reply:
i tried to assert attribute ids, but then dropped (because actually the nmae seems to be enough). Forgot to delete it.